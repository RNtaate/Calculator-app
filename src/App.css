*,
*::after, 
*::before {
  box-sizing: border-box;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.main-div {
  background-color: #123456;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.display-input {
  background-color: #777;
  color: white;
  width: 700px;
  height: 100px;
  border:none;
  outline: none;
  font-size: 60px;
  font-weight: bold;
  font-family: 'Times New Roman', Times, serif;
  padding: 0 20px;
}

.calculator-panel {
  width: 700px;
  background-color: black;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.calculator-panel > div {
  display: flex;
  width: 100%;
  height: 100px;
}

.calculator-panel > div > button {
  font-size: 25px;
  font-weight: bold;
  border: 1px solid #999;
  background-color: #FF893C;
}