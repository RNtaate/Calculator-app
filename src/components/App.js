import React, { useState } from 'react';
import '../App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

const App = () => {
  const [calcData, setCalcData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    let tempData = { ...calcData };
    tempData = calculate(buttonName, tempData);
    setCalcData(tempData);
  };

  const updateDisplayResult = () => {
    let displayResult = '';
    let { total, next: myNext, operation } = calcData;
    if ((total === null) && (myNext === null)) {
      displayResult = '0';
    } else {
      total = total === null ? '' : total;
      myNext = myNext === null ? '' : myNext;
      operation = operation === null ? '' : operation;

      displayResult = `${total} ${operation} ${myNext}`;
    }

    return displayResult;
  };

  return (
    <>
      <Display result={updateDisplayResult()} />
      <ButtonPanel clickHandler={handleClick} />
    </>
  );
};

export default App;
