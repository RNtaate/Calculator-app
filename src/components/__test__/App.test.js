import { render, screen } from '@testing-library/react';
import App from '../App';

import renderer from 'react-test-renderer';

test('renders App component correctly', () => {
  render(<App />);
  const linkElement = screen.getByText(/AC/i);
  expect(linkElement).toBeInTheDocument();
});

test('matches App component snapshot', () => {
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
});