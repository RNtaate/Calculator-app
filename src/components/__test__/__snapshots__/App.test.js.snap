// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches App component snapshot 1`] = `
<div
  className="main-div"
>
  <input
    className="display-input"
    readOnly={true}
    style={
      Object {
        "textAlign": "right",
      }
    }
    type="text"
    value="0"
  />
  <div
    className="calculator-panel"
  >
    <div
      className="group1"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="%"
      >
        %
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "width": "25%",
          }
        }
        type="button"
        value="รท"
      >
        รท
      </button>
    </div>
    <div
      className="group2"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="7"
      >
        7
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="8"
      >
        8
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="9"
      >
        9
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "width": "25%",
          }
        }
        type="button"
        value="X"
      >
        X
      </button>
    </div>
    <div
      className="group3"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="4"
      >
        4
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="5"
      >
        5
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="6"
      >
        6
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "width": "25%",
          }
        }
        type="button"
        value="-"
      >
        -
      </button>
    </div>
    <div
      className="group4"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="1"
      >
        1
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="2"
      >
        2
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="3"
      >
        3
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "width": "25%",
          }
        }
        type="button"
        value="+"
      >
        +
      </button>
    </div>
    <div
      className="group5"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "50%",
          }
        }
        type="button"
        value="0"
      >
        0
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "width": "25%",
          }
        }
        type="button"
        value="."
      >
        .
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "width": "25%",
          }
        }
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;
